pet.el

```diff
@@ -923,15 +911,74 @@ FN is `eglot--guess-contact', ARGS is the arguments to
                 (seq-subseq result 4))
       result)))
 
+(defun pet-eglot-alternatives (alternatives)
+  "Use `pet-executable-find' instead of `executable-find' in the
+`eglot-alternatives' closure."
+  (lambda (&optional interactive _project)
+    ;; JT@2021-06-13: This function is way more complicated than it
+    ;; could be because it accounts for the fact that Compat's
+    ;; `executable-find' may take much longer to execute on
+    ;; remote files.
+    (let* ((listified (cl-loop for a in alternatives
+                               collect (if (listp a) a (list a))))
+           (err (lambda ()
+                  (error "None of '%s' are valid executables"
+                         (mapconcat #'car listified ", ")))))
+      (cond ((and interactive current-prefix-arg)
+             ;; A C-u always lets user input something manually,
+             nil)
+            (interactive
+             (let* ((augmented (mapcar (lambda (a)
+                                         (let ((found (pet-executable-find (car a))))
+                                           (and found
+                                                (cons (car a) (cons found (cdr a))))))
+                                       listified))
+                    (available (remove nil augmented)))
+               (cond ((cdr available)
+                      (cdr (assoc
+                            (completing-read
+                             "[eglot] More than one server executable available: "
+                             (mapcar #'car available)
+                             nil t nil nil (car (car available)))
+                            available #'equal)))
+                     ((cdr (car available)))
+                     (t
+                      ;; Don't error when used interactively, let the
+                      ;; Eglot prompt the user for alternative (github#719)
+                      nil))))
+            (t
+             (cl-loop for (p . args) in listified
+                      for probe = (pet-executable-find p)
+                      when probe return (cons probe args)
+                      finally (funcall err)))))))
+
+(defun pet-eglot--lookup-mode-advice (fn &rest args)
+  "Advice `eglot--lookup-mode' to use `pet-executable-find' if in Python
+mode.
+
+FN is `eglot--lookup-mode', ARGS is the arguments to
+`eglot--lookup-mode'."
+  (if (provided-mode-derived-p (car args)
+                               (if (functionp 'python-base-mode)
+                                   '(python-base-mode) '(python-mode)))
+      (let ((eglot-server-programs `(((python-mode python-ts-mode)
+                                      ;; Based on eglot v1.17, 2024-12-12
+                                      . ,(pet-eglot-alternatives
```

Comment:
If you advice `eglot--lookup-mode`, you can just look up the executables here right? Why do you still need to advice `eglot-alternatives`?